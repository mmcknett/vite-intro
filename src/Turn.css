.correct {
  color: green;
}

.misplaced {
  color: yellow;
}

.incorrect {
  color: inherit;
}
