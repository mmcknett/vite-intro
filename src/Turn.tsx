import './Turn.css';

type TurnProps = {
  guess: string,
  referenceWord: string
}
const Turn: React.FC<TurnProps> = ({guess, referenceWord, }) => {
  const guessList = guess.split("");
  const referenceWordList = referenceWord.split("");
  // if Item is in the same position => Green
  // If Item is in list but incorrect position => Yellow
  // If Item is not in list => Red
  // [["A", "correct"], ["N", "incorrect"], ["G", "incorrect"], ["L", "incorrect"], ["E", "misplaced"]]
  const results = guessList.map((guessLetter, index) => {
    if(guessLetter === referenceWordList[index]){
      return [guessLetter, 'correct']
    } else if(referenceWordList.includes(guessLetter)){
      return [guessLetter, 'misplaced']
    } else {
      return [guessLetter, 'incorrect']
    }
  })


  return (
    <>
      { results.map((result, index) => (<span key={index} className={result[1]} data-testid={`${guess}-${index}`}>{result[0]}</span>)) }
    </>
  );
}

export default Turn;