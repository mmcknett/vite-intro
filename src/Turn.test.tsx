import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import Turn from './Turn.jsx';

describe('Turn', () => {
  it('renders with no props as expected', () => {
    const { asFragment } = render(< Turn guess="Angle" referenceWord="Amber" />);
    expect(asFragment()).toMatchSnapshot();
  })

  it('renders the success css class correctly', () => {
    const { queryAllByTestId } = render(< Turn guess="Angle" referenceWord="Amber" />);
    expect(queryAllByTestId("Angle-0")[0].classList.contains('correct')).toBe(true);
  })
});