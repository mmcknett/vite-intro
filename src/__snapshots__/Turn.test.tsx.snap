// Vitest Snapshot v1

exports[`Turn > renders with no props as expected 1`] = `
<DocumentFragment>
  <span
    class="correct"
    data-testid="Angle-0"
  >
    A
  </span>
  <span
    class="incorrect"
    data-testid="Angle-1"
  >
    n
  </span>
  <span
    class="incorrect"
    data-testid="Angle-2"
  >
    g
  </span>
  <span
    class="incorrect"
    data-testid="Angle-3"
  >
    l
  </span>
  <span
    class="misplaced"
    data-testid="Angle-4"
  >
    e
  </span>
</DocumentFragment>
`;
